seaport-subgraph/src/seaport.ts

```diff
@@ -0,0 +1,220 @@
+
+          let sale = new airstack.nft.Sale(buyer, seller, nft, paymentAmount, paymentToken, protocolFees, protocolBeneficiary, new Array<airstack.nft.CreatorRoyalty>());
+          allSales.push(sale);
+          
+          log.info(
+            "txHash consideration log tx {} logindex {} nftContract {} NFTId {} index {} recipient {} buyer {} seller {}",
+            [
+              txHash.toHexString(),
+              event.logIndex.toString(),
+              consideration.token.toHexString(),
+              consideration.identifier.toString(),
+              i.toString(),
+              consideration.recipient.toHexString(),
+              buyer.toHexString(),
+              seller.toHexString(),
+            ]
+          );
+        }
+    }
+    
+    for(let i = 0; i <allSales.length; i++){
+        let royalty = new airstack.nft.CreatorRoyalty(royaltyFees.div(BigInt.fromI64(allSales.length)), royaltyBeneficiary);
+        allSales[i].royalties.push(royalty);
+        allSales[i].protocolFees = protocolFees.div(BigInt.fromI64(allSales.length));
+        allSales[i].protocolFeesBeneficiary = protocolBeneficiary;
+        allSales[i].paymentAmount = paymentAmount.div(
+          BigInt.fromI32(allSales.length)
+        ); // For bundle sale, equally divide the payment amount in all sale transaction
+        allSales[i].paymentToken = paymentToken;
+        log.info("txHash {} royaltyBeneficiary {} feeBeneficiary {} feeAmount {}",
+          [ 
+            txHash.toHexString(),
+            royaltyBeneficiary.toHexString(),
+            protocolBeneficiary.toHexString(),
+            // allSales[i].royalties[0].beneficiary.toHexString(),
+            allSales[i].protocolFees.toString(),
+          ]
+        )
+    }
+    log.info("txHash {} allSales length {}", [txHash.toHexString(), allSales.length.toString()]);
+    if (allSales.length > 0) {
+      if (allSales[0].seller == Address.zero() || allSales[0].buyer == Address.zero()){
+        let token_exist = WrappedEtherTransaction.load("mainnet-"+txHash.toHexString()+"-"+allSales[0].nft.collection.toHexString() + allSales[0].nft.tokenId.toString());
+        if (token_exist == null){
+          token_exist = new WrappedEtherTransaction("mainnet-"+txHash.toHexString()+"-"+allSales[0].nft.collection.toHexString() + allSales[0].nft.tokenId.toString());
+          token_exist.from = allSales[0].seller.toHexString();
+          token_exist.to = allSales[0].buyer.toHexString();
+          token_exist.hash = txHash.toHexString();
+          token_exist.save();
+        }else{
+          if (allSales[0].seller == Address.zero()){
+            allSales[0].seller = Address.fromString(token_exist.from);
+          }
+        }
+      }
+    }
+    airstack.nft.trackNFTSaleTransactions(
+      "1",
+      txHash.toHexString(),
+      event.transaction.index,
+      allSales,
+      "NFT_MARKET_PLACE",
```

Comment:
Please create constants for all the string.

Reply:
Done